gamestate.js

```diff
@@ -0,0 +1,99 @@
+function toCost(text) {
+    var cost = "&nbsp;";
+    if (text) {
+        for (var i = 0; i < text.length; i++) {
+            cost += "<i class='ms ms-" + text[i] + "'></i>";
+        }
+    }
+    return cost;
+}
+
+function fillHand(handElt, deck) {
+    handElt.empty();
+    for (var i = 0; i < deck.length; i++) {
+        var card = deck[i];
+        if (card.inhand) {
+            var cardElt = $("<div>", {"class": "card"});
+            cardElt.text(card.name);
+            var costElt = $("<span>", {"class": "mana"});
+            costElt.html(toCost(card.cost && ("" + card.cost)));
+            cardElt.append(costElt);
+            handElt.append(cardElt)
+        }
+    }
+}
+
+// Initialize Firebase
+firebase.initializeApp(config);
+firebase.auth().signInWithEmailAndPassword(firebaseUsername, firebasePassword).catch(function (error) {
+    console.log(error);
+});
+
+var modeRef = firebase.database().ref('mode');
+console.log(modeRef);
+modeRef.on('value', function (v) {
+    mode = v.val();
+    console.log(mode);
+    modes = ["sideboard", "game", "title", "freetext"];
+    _.each(modes, function (ele, idx, list) {
+        console.log(ele);
+        $("#" + ele).hide();
+    });
+    if (_.contains(modes, mode)) {
+        console.log(mode);
+        $("#" + mode).show();
+    }
+});
+firebase.database().ref('player1').on('value', function (v) {
+    console.log(v.val());
+    $(".left .life").text(v.val().life);
+    var poison = v.val().poison;
+    var poisonElt = $(".left .poison");
+    poisonElt.text(v.val().poison);
+    poisonElt.toggle(!!v.val().poison);
+    $("#p1name").text(v.val().name);
+    $("#p1deck").text(v.val().deck);
+    $("#p1wins").text(v.val().gamewins);
+    $(".sideboard .left").text(v.val().sideboard.replace(/_/g, "\n"));
+});
+firebase.database().ref('p1deck').on('value', function (v) {
+    fillHand($("#p1hand"), v.val());
+});
+firebase.database().ref('player2').on('value', function (v) {
+    console.log(v.val());
+    $(".right .life").text(v.val().life);
+    var poison = v.val().poison;
+    var poisonElt = $(".right .poison");
```

Comment:
so when you did this were you just following my conventions or did what I did actually make sense